src/zalasr.adoc

```diff
@@ -2,11 +2,31 @@
 
 The Zalasr extension provides load-acquire and store-release instructions in RISC-V.
 These can be important for high performance designs by enabling finer-grained synchronisation than is possible with fences alone, by providing a unidirectional fence.
-Load-acquire and store-release are also widely used in language-level memory models.
-Both the Java and {cpp} memory models make use of acquire-release semantics, and {cpp}'s `atomic` provides primitives that are meant to map directly to load-acquire and store-release instructions.
+Load-acquire and store-release are widely used in language-level memory models:
+both the Java and {cpp} memory models make use of acquire-release semantics, and {cpp}'s `atomic` provides primitives that are meant to map directly to load-acquire and store-release instructions.
+
+The Zalasr extension builds on the atomic support provided by the Zaamo, Zalrsc, and Zabha extensions by providing additional atomic operations (although it does not depend on them).
+All of the AMO operations in Zaamo (and Zabha) are read-modify-write operations that both load and store.
+Zalasr allows for using only a load xor store which reduces the ordering requirements and requires less cache traffic.
+The Zalrsc extension provides operations that are only loads xor stores.
+However, since it is designed to perform an atomic operation on a single address, the loads and stores are designed to be paired.
+The load-reserved implies an intent to modify, leading to additional cache traffic, and store-conditional requires that there was a previous load-reserved without other intervening loads or stores.
+Therefore, the Zalrsc extension does not provide a general atomic and ordered load or store.
+
+Zalasr fills this gap by offering truly standalone atomic and ordered loads and stores.
+The Zalasr instructions are atomic loads and stores that support ordering annotations.
+With the combination of Zaamo, Zabha, Zalasr, and Zalrsc all {cpp} atomic operations can be supported with a single instruction.
```

Comment:
Zalrsc by defintion is a multiple instruction sequence. 
"with a single instruction" -> "with single instructions"